### PR TITLE
Remove mtls.enabled=true CRs

### PR DESCRIPTION
**Pull request by howardjohn**
We are now recommending users to manage these directly instead of
relying on installation

### CODE REVIEW

**Comment by howardjohn** [comment]
This is broken out of https://github.com/istio/istio/pull/22056

File: manifests/istio-control/istio-discovery/templates/enable-mesh-mtls.yaml [hunk 1]
```diff
@@ -1,51 +0,0 @@
-{{- /*
-
-TODO(https://github.com/istio/istio/issues/18199) remove this configuration from charts once the operator starts managing it
-
-*/ -}}
-
-{{ if .Values.clusterResources }}
-{{- if .Values.global.mtls.enabled }}
```
Comment (elfinhe) [inline manifests/istio-control/istio-discovery/templates/enable-mesh-mtls.yaml]: Does this mean someone need to add those CRs into bookinfo example and other demos?

Comment (howardjohn) [reply manifests/istio-control/istio-discovery/templates/enable-mesh-mtls.yaml]: I believe @diemtvu already updated all the docs

**Review by elfinhe — APPROVED** [review]

File: manifests/istio-control/istio-discovery/templates/enable-mesh-mtls.yaml (hunk 1 above)
Comment (elfinhe) [reply manifests/istio-control/istio-discovery/templates/enable-mesh-mtls.yaml]: I https://istio.io/docs/examples/bookinfo/#apply-default-destination-rules

`$ kubectl apply -f samples/bookinfo/networking/destination-rule-all-mtls.yaml`

must cover the one in bookinfo.

**Review by diemtvu — APPROVED** [review]

**Pull request merged**